core/src/main/scala/org/dbpedia/extraction/mappings/wikidata/WikidataR2RExtractor.scala

```diff
@@ -219,18 +230,49 @@ class WikidataR2RExtractor(
 
     value match {
       case v: ItemIdValue => {
-        val valueTitle = "wikidata:" + WikidataUtil.getItemId(v)
-        context.ontology.wikidataClassesMap.foreach({ map =>
-          if (map._1.matches(valueTitle)) {
-            classes ++= map._2
+        val wikidataItem = WikidataUtil.getItemId(v)
+          classMappings.get(wikidataItem) match {
+            case Some(mappings) => classes++=mappings
+            case _=>
           }
-        })
       }
       case _ =>
     }
     classes
   }
 
+  private def readClassMappings(fileName:String): mutable.Map[String,Set[OntologyClass]] = {
+    val finalMap = mutable.Map[String,Set[OntologyClass]]()
+
+    try {
+      val source = scala.io.Source.fromFile(fileName)
+      val jsonString = source.getLines() mkString
+      val mapper = new ObjectMapper() with ScalaObjectMapper
+      mapper.registerModule(DefaultScalaModule)
+      val mapFromJson = mapper.readValue[Map[String, String]](jsonString)
+
+      mapFromJson.foreach{
+        pair =>
+          mapFromJson.get(pair._1) match {
+            case Some(ontologyKey) => {
+              context.ontology.classes.get(ontologyKey) match {
+                case Some(oClass)=> finalMap+=pair._1 -> Set(oClass)
+                case _=>
+              }}
+            case _=>
+          }
+      }
+    } catch {
+      case ioe: IOException => println("Please check class mapping file "+ioe)
+    }
+
+    context.ontology.wikidataClassesMap.foreach {
```

Comment:
we should check if the key exists before, if not, we keep as is, if it exists we combine the values.
This does not change any of the current results but if someone runs with newer version of the ontology and old mappings file we might get collisions